src/github.com/matrix-org/rageshake/submit.go

```diff
@@ -93,12 +95,20 @@ func (s *submitServer) saveReport(ctx context.Context, p payload) error {
 	//  "bugreport-20170115-112233-N.log.gz" => oldest log
 	t := time.Now().UTC()
 	prefix := t.Format("2006-01-02/150405")
-	summary := fmt.Sprintf(
-		"%s\n\nNumber of logs: %d\nVersion: %s\nUser-Agent: %s\n", p.Text, len(p.Logs), p.Version, p.UserAgent,
+
+	var summaryBuf bytes.Buffer
+	fmt.Fprintf(
+		&summaryBuf,
+		"%s\n\nNumber of logs: %d\nApplication: %s\nVersion: %s\nUser-Agent: %s\n",
```

Comment:
I'm surprised this actually works as intended. Yes, you should use `bytes.Buffer` to efficiently concatenate strings, but there is a specific `buf.WriteString(str)` function you should use, and then `buf.String()` to get the complete output. That would be much clearer, though you'll still need to `fmt.Sprintf` to format the string.